app-typescript/components/DatePicker.tsx

```diff
@@ -160,7 +161,8 @@ export class DatePicker extends React.PureComponent<IDatePicker, IState> {
                 tabindex={this.props.tabindex}
             >
                 <Calendar
-                    footerTemplate={this.props.required !== true ? () => (
+                    className='sd-input__input'
+                    footerTemplate={this.props.hideClear !== true ? () => (
```

Comment:
maintain logic regarding "required"

Reply:
This was a choice I made, because I didn't want to bloat the logic. I chose to bloat the interface instead and require users of the component to have to also pass `hideClear`. There's also a benefit to this I think - I don't couple the required logic with hiding the clear when you don't want it for other reasons.

Reply:
It's a violation of a code contract between interface(props) and implementation. As a developer I say via props that field must not be empty, and you allow it to be empty - ignoring what was requested via props.


Bloat is a secondary issue compared to this. I don't see it as bloat either - being able to clear the value is a core function of any input.